ksz8863/src/ksz8863_ctrl.c

```diff
@@ -57,56 +47,27 @@ static inline bool bus_unlock(void)
 static esp_err_t ksz8863_i2c_write(uint8_t reg_addr, uint8_t *data, size_t len)
 {
     esp_err_t ret = ESP_OK;
-    i2c_port_t i2c_port = s_ksz8863_ctrl_intf->i2c_bus_spec.i2c_port;
-    uint8_t dev_addr = s_ksz8863_ctrl_intf->i2c_bus_spec.dev_addr;
-
-    i2c_cmd_handle_t cmd = i2c_cmd_link_create();
-    ESP_RETURN_ON_FALSE(cmd, ESP_ERR_NO_MEM, TAG, "I2C link create error");
-    ESP_GOTO_ON_ERROR(i2c_master_start(cmd), err, TAG, "I2C master start error");
-    ESP_GOTO_ON_ERROR(i2c_master_write_byte(cmd, dev_addr | I2C_MASTER_WRITE, true), err, TAG, "I2C master write error");
-    ESP_GOTO_ON_ERROR(i2c_master_write_byte(cmd, reg_addr, true), err, TAG, "I2C master write error");
-    if (likely(reg_addr != KSZ8863_RESET_ADDR)) {
-        ESP_GOTO_ON_ERROR(i2c_master_write(cmd, data, len, true), err, TAG, "I2C master write error");
+    uint8_t *reg_addr_and_data = malloc(len + 1);
+    // Create a packet containing the register and data to be transmitted
+    reg_addr_and_data[0] = reg_addr;
+    memcpy(reg_addr_and_data + 1, data, len);
+    // When performing a soft reset, the KSZ8863 doesn't produce an ACK. Print a warning that the error is expected and ignore it.
+    if unlikely(reg_addr == KSZ8863_RESET_ADDR) {
+        ESP_LOGW(TAG, "Ignore the error below. KSZ8863 does not produce ACK when performing soft reset.");
```

Comment:
It needs to be specifically said it's I2C error to ignore.

Reply:
Rephrased. Also added clarification of why we won't support SMI access to the README.